astropy/io/fits/tests/test_hdulist.py

```diff
@@ -178,6 +178,39 @@ def test_append_groupshdu_to_non_empty_list(self):
         with pytest.raises(ValueError):
             hdul.append(hdu)
 
+    @pytest.mark.parametrize(
+        ["image", "do_not_scale"],
+        [
+            ["scale.fits", True],
+            ["o4sp040b0_raw.fits", True],
+            ["fixed-1890.fits", True],
+            ["scale.fits", False],
+            ["o4sp040b0_raw.fits", False],
+            ["fixed-1890.fits", False],
+        ],
+    )
+    def test_append_scaled_image_with_do_not_scale_image_data(
+        self, image, do_not_scale
+    ):
+        """Tests appending a scaled ImageHDU to a HDUList."""
+
+        with (
+            fits.open(self.data(image), do_not_scale_image_data=do_not_scale) as source,
+            fits.open(self.temp("dest.fits"), mode="append") as dest,
```

Comment:
"dest.fits" is not used below since you are writing the file with `dest.writeto` so you can just create the HDUList below (`dest = fits.HDUList()`).